docs/fault-proof-alpha/immunifi.md

```diff
@@ -0,0 +1,82 @@
+1. Including an extra layer of bisection over output roots, enabling the off-chain challenge agents to only need to run [Cannon][cannon] over a single block rather than a string of blocks. This will heavily reduce the hardware cost of running the off
+   chain challenge agent, which mitigates the DoS vector mentioned above.
+1. Adding bonds to the system to preserve incentive compatibility. In the alpha, defenses of the honest L2 state are not incentivized, which also means that attacks on the honest L2 state are not disincentivized. Adding bonds to each claim
+   made in the dispute game will preserve the incentives of the system as well as make it more costly to attack.
+1. Improving the [Dispute Game][dispute-game]'s resolution algorithm to reduce the number of interactions that the off-chain challenge agents need to have with the on-chain dispute game. This will reduce the cost of running the off-chain challenge
+   agent, ensure that an honest challenger's participation always results in a profitable move, and possibly prevent the need for challengers to respond to every invalid claim within the game.
+1. The fault proof system will be integrated into the bridge contracts, specifically the `OptimismPortal`, in order to enable the system to be used in production and verify the correctness of output roots that withdrawals are proven against.
+
+## Bounty Scope
+The scope of the bounty is limited to the fault proof alpha system. This includes the following components, in order of security review priority:
+1. **Cannon**: The [Cannon][cannon] binary and its dependencies.
+1. **op-program**: The [`op-program`][op-program] binary and its dependencies.
+1. **Smart Contracts**
+    1. The [Cannon][cannon-contracts] contracts and their dependencies.
+    1. The [Dispute Game][dispute-game] and their dependencies.
+1. **op-challenger**: The [`op-challenger`][op-challenger] binary and its dependencies.
+
+As mentioned above in the "[Plans for the next iteration](#plans-for-the-next-iteration)" section, there will soon be a number of large architectural changes to the [dispute smart contracts][dispute-game]
+as well as the [`op-challenger`][op-challenger] in order to support the features that will bring the system to a production ready state. During this time, it is unlikely that [Cannon][cannon], the [Cannon contracts][cannon-contracts],
+or the [`op-program`][op-program] will change significantly, and as such, we recommend focusing efforts primarily on these components.
+
+There are several key invariants that must be maintained in order for the system to be considered secure:
+1. **Cannon**
+    1. [Cannon][cannon]'s `mipsevm` must be functionally equivalent to the [MIPS thread context][cannon-contracts] implemented in Solidity. Any disparities in behavior is considered a bug.
```

Comment:
Any disparity or any disparity that affects op-program execution?

Reply:
Updated this - good callout. Let me know if this needs more explanation :)